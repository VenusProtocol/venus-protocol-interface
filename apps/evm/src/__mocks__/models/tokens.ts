import bnbLogo from 'packages/tokens/img/bnb.svg';
import busdLogo from 'packages/tokens/img/busd.svg';
import ethLogo from 'packages/tokens/img/eth.svg';
import lisUsdLogo from 'packages/tokens/img/lisUSD.png';
import lunaLogo from 'packages/tokens/img/luna.svg';
import usdcLogo from 'packages/tokens/img/usdc.svg';
import usdtLogo from 'packages/tokens/img/usdt.svg';
import ustLogo from 'packages/tokens/img/ust.svg';
import vaiLogo from 'packages/tokens/img/vai.svg';
import vrtLogo from 'packages/tokens/img/vrt.svg';
import wbnbLogo from 'packages/tokens/img/wbnb.svg';
import xvsLogo from 'packages/tokens/img/xvs.svg';
import { Token } from 'types';

export const xvs: Token = {
  address: '0xB9e0E753630434d7863528cc73CB7AC638a7c8ff',
  decimals: 18,
  symbol: 'XVS',
  asset: xvsLogo,
};

export const vai: Token = {
  address: '0x5fFbE5302BadED40941A403228E6AD03f93752d9',
  decimals: 18,
  symbol: 'VAI',
  asset: vaiLogo,
};

export const vrt: Token = {
  address: '0x331F639B4F3CF6E391CD244e0b5027C5968Ec325',
  decimals: 18,
  symbol: 'VRT',
  asset: vrtLogo,
};

export const bnb: Token = {
  address: '0x0000000000000000000000000000000000000000',
  isNative: true,
  decimals: 18,
  symbol: 'BNB',
  asset: bnbLogo,
};

export const usdc: Token = {
  address: '0x16227D60f7a0e586C66B005219dfc887D13C9531',
  decimals: 6,
  symbol: 'USDC',
  asset: usdcLogo,
};

export const usdt: Token = {
  address: '0xA11c8D9DC9b66E209Ef60F0C8D969D3CD988782c',
  decimals: 6,
  symbol: 'USDT',
  asset: usdtLogo,
};

export const busd: Token = {
  address: '0x8301F2213c0eeD49a7E28Ae4c3e91722919B8B47',
  decimals: 18,
  symbol: 'BUSD',
  asset: busdLogo,
};

export const ust: Token = {
  address: '0x5A79efD958432E72211ee73D5DDFa9bc8f248b5F',
  decimals: 18,
  symbol: 'UST',
  asset: ustLogo,
};

export const luna: Token = {
  address: '0xf36160EC62E3B191EA375dadfe465E8Fa1F8CabB',
  decimals: 6,
  symbol: 'LUNA',
  asset: lunaLogo,
};

export const wbnb: Token = {
  symbol: 'WBNB',
  decimals: 18,
  address: '0xae13d989daC2f0dEbFf460aC112a837C89BAa7cd',
  asset: wbnbLogo,
};

export const eth: Token = {
  address: '0x98f7A83361F7Ac8765CcEBAB1425da6b341958a7',
  decimals: 18,
  symbol: 'ETH',
  asset: ethLogo,
};

export const lisUsd: Token = {
  address: '0xe73774DfCD551BF75650772dC2cC56a2B6323453',
  decimals: 18,
  symbol: 'lisUSD',
  asset: lisUsdLogo,
};

export default [xvs, bnb, usdc, usdt, busd, ust, luna, vai, wbnb, eth, vrt, lisUsd];
