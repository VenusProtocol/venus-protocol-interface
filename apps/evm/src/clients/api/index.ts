export { default as queryClient } from './queryClient';

// Mutations
export { default as mintVai } from './mutations/mintVai';
export * from './mutations/mintVai';
export { default as useMintVai } from './mutations/mintVai/useMintVai';

export { default as repayVai } from './mutations/repayVai';
export * from './mutations/repayVai';
export { default as useRepayVai } from './mutations/repayVai/useRepayVai';

export { default as enterMarket } from './mutations/enterMarket';
export * from './mutations/enterMarket';
export { default as useEnterMarket } from './mutations/enterMarket/useEnterMarket';

export { default as exitMarket } from './mutations/exitMarket';
export * from './mutations/exitMarket';
export { default as useExitMarket } from './mutations/exitMarket/useExitMarket';

export { default as approveToken } from './mutations/approveToken';
export * from './mutations/approveToken';
export { default as useApproveToken } from './mutations/approveToken/useApproveToken';

export { default as revokeSpendingLimit } from './mutations/revokeSpendingLimit';
export * from './mutations/revokeSpendingLimit';
export { default as useRevokeSpendingLimit } from './mutations/revokeSpendingLimit/useRevokeSpendingLimit';

export { default as supply } from './mutations/supply';
export * from './mutations/supply';
export { default as useSupply } from './mutations/supply/useSupply';
export * from './mutations/supply/useSupply';

export { default as redeem } from './mutations/redeem';
export * from './mutations/redeem';
export { default as useRedeem } from './mutations/redeem/useRedeem';

export { default as repay } from './mutations/repay';
export * from './mutations/repay';
export { default as useRepay } from './mutations/repay/useRepay';

export { default as redeemUnderlying } from './mutations/redeemUnderlying';
export * from './mutations/redeemUnderlying';
export { default as useRedeemUnderlying } from './mutations/redeemUnderlying/useRedeemUnderlying';

export { default as borrow } from './mutations/borrow';
export * from './mutations/borrow';
export { default as useBorrow } from './mutations/borrow/useBorrow';

export { default as withdrawXvs } from './mutations/withdrawXvs';
export * from './mutations/withdrawXvs';
export { default as useWithdrawXvs } from './mutations/withdrawXvs/useWithdrawXvs';

export { default as setVoteDelegate } from './mutations/setVoteDelegate';
export * from './mutations/setVoteDelegate';
export { default as useSetVoteDelegate } from './mutations/setVoteDelegate/useSetVoteDelegate';

export { default as createProposal } from './mutations/createProposal';
export * from './mutations/createProposal';
export { default as useCreateProposal } from './mutations/createProposal/useCreateProposal';

export { default as cancelProposal } from './mutations/cancelProposal';
export * from './mutations/cancelProposal';
export { default as useCancelProposal } from './mutations/cancelProposal/useCancelProposal';

export { default as executeProposal } from './mutations/executeProposal';
export * from './mutations/executeProposal';
export { default as useExecuteProposal } from './mutations/executeProposal/useExecuteProposal';

export { default as queueProposal } from './mutations/queueProposal';
export * from './mutations/queueProposal';
export { default as useQueueProposal } from './mutations/queueProposal/useQueueProposal';

export { default as stakeInXvsVault } from './mutations/stakeInXvsVault';
export * from './mutations/stakeInXvsVault';
export { default as useStakeInXvsVault } from './mutations/stakeInXvsVault/useStakeInXvsVault';

export { default as stakeInVaiVault } from './mutations/stakeInVaiVault';
export * from './mutations/stakeInVaiVault';
export { default as useStakeInVaiVault } from './mutations/stakeInVaiVault/useStakeInVaiVault';

export { default as castVote } from './mutations/vote/castVote';
export * from './mutations/vote/castVote';
export { default as useCastVote } from './mutations/vote/useCastVote';

export { default as castVoteWithReason } from './mutations/vote/castVoteWithReason';
export * from './mutations/vote/castVoteWithReason';
export { default as useCastVoteWithReason } from './mutations/vote/useCastVoteWithReason';

export { default as withdrawFromVaiVault } from './mutations/withdrawFromVaiVault';
export * from './mutations/withdrawFromVaiVault';
export { default as useWithdrawFromVaiVault } from './mutations/withdrawFromVaiVault/useWithdrawFromVaiVault';

export { default as requestWithdrawalFromXvsVault } from './mutations/requestWithdrawalFromXvsVault';
export * from './mutations/requestWithdrawalFromXvsVault';
export { default as useRequestWithdrawalFromXvsVault } from './mutations/requestWithdrawalFromXvsVault/useRequestWithdrawalFromXvsVault';

export { default as executeWithdrawalFromXvsVault } from './mutations/executeWithdrawalFromXvsVault';
export * from './mutations/executeWithdrawalFromXvsVault';
export { default as useExecuteWithdrawalFromXvsVault } from './mutations/executeWithdrawalFromXvsVault/useExecuteWithdrawalFromXvsVault';

export { default as swapTokens } from './mutations/swapTokens';
export * from './mutations/swapTokens';
export { default as useSwapTokens } from './mutations/swapTokens/useSwapTokens';

export { default as swapTokensAndRepay } from './mutations/swapTokensAndRepay';
export * from './mutations/swapTokensAndRepay';
export { default as useSwapTokensAndRepay } from './mutations/swapTokensAndRepay/useSwapTokensAndRepay';

export { default as swapTokensAndSupply } from './mutations/swapTokensAndSupply';
export * from './mutations/swapTokensAndSupply';
export { default as useSwapTokensAndSupply } from './mutations/swapTokensAndSupply/useSwapTokensAndSupply';

export { default as claimRewards } from './mutations/claimRewards';
export * from './mutations/claimRewards';
export { default as useClaimRewards } from './mutations/claimRewards/useClaimRewards';

export { default as useStakeInVault } from './mutations/useStakeInVault';

export { default as claimPrimeToken } from './mutations/claimPrimeToken';
export * from './mutations/claimPrimeToken';
export { default as useClaimPrimeToken } from './mutations/claimPrimeToken/useClaimPrimeToken';

export { default as bridgeXvs } from './mutations/bridgeXvs';
export * from './mutations/bridgeXvs';
export { default as useBridgeXvs } from './mutations/bridgeXvs/useBridgeXvs';

export { default as wrapTokensAndSupply } from './mutations/wrapTokensAndSupply';
export * from './mutations/wrapTokensAndSupply';
export { default as useWrapTokensAndSupply } from './mutations/wrapTokensAndSupply/useWrapTokensAndSupply';

export { default as updatePoolDelegateStatus } from './mutations/updatePoolDelegateStatus';
export * from './mutations/updatePoolDelegateStatus';
export { default as useUpdatePoolDelegateStatus } from './mutations/updatePoolDelegateStatus/useUpdatePoolDelegateStatus';

export { default as borrowAndUnwrap } from './mutations/borrowAndUnwrap';
export * from './mutations/borrowAndUnwrap';
export { default as useBorrowAndUnwrap } from './mutations/borrowAndUnwrap/useBorrowAndUnwrap';

export { default as redeemAndUnwrap } from './mutations/redeemAndUnwrap';
export * from './mutations/redeemAndUnwrap';
export { default as useRedeemAndUnwrap } from './mutations/redeemAndUnwrap/useRedeemAndUnwrap';

export { default as redeemUnderlyingAndUnwrap } from './mutations/redeemUnderlyingAndUnwrap';
export * from './mutations/redeemUnderlyingAndUnwrap';
export { default as useRedeemUnderlyingAndUnwrap } from './mutations/redeemUnderlyingAndUnwrap/useRedeemUnderlyingAndUnwrap';

// Queries
export { default as getVaiCalculateRepayAmount } from './queries/getVaiCalculateRepayAmount';
export * from './queries/getVaiCalculateRepayAmount';
export { default as useGetVaiCalculateRepayAmount } from './queries/getVaiCalculateRepayAmount/useGetVaiCalculateRepayAmount';

export { default as getVaiRepayAmountWithInterests } from './queries/getVaiRepayAmountWithInterests';
export * from './queries/getVaiRepayAmountWithInterests';
export { default as useGetVaiRepayAmountWithInterests } from './queries/getVaiRepayAmountWithInterests/useGetVaiRepayAmountWithInterests';

export { default as getVaiTreasuryPercentage } from './queries/getVaiTreasuryPercentage';
export * from './queries/getVaiTreasuryPercentage';
export { default as useGetVaiTreasuryPercentage } from './queries/getVaiTreasuryPercentage/useGetVaiTreasuryPercentage';

export { default as getHypotheticalAccountLiquidity } from './queries/getHypotheticalAccountLiquidity';
export * from './queries/getHypotheticalAccountLiquidity';

export { default as getLegacyPoolMarkets } from './queries/getLegacyPoolMarkets';
export * from './queries/getLegacyPoolMarkets';
export { default as useGetLegacyPoolMarkets } from './queries/getLegacyPoolMarkets/useGetLegacyPoolMarkets';

export { default as getVTokenBalancesAll } from './queries/getVTokenBalancesAll';
export * from './queries/getVTokenBalancesAll';
export { default as useGetVTokenBalancesAll } from './queries/getVTokenBalancesAll/useGetVTokenBalancesAll';

export { default as getVTokenBalanceOf } from './queries/getVTokenBalanceOf';
export * from './queries/getVTokenBalanceOf';
export { default as useGetVTokenBalanceOf } from './queries/getVTokenBalanceOf/useGetVTokenBalanceOf';

export { default as getPendingRewards } from './queries/getPendingRewards';
export * from './queries/getPendingRewards';
export { default as useGetPendingRewards } from './queries/getPendingRewards/useGetPendingRewards';

export { default as getIsAddressAuthorized } from './queries/getIsAddressAuthorized';
export * from './queries/getIsAddressAuthorized';
export { default as useGetIsAddressAuthorized } from './queries/getIsAddressAuthorized/useGetIsAddressAuthorized';

export { default as getAllowance } from './queries/getAllowance';
export * from './queries/getAllowance';
export { default as useGetAllowance } from './queries/getAllowance/useGetAllowance';

export { default as getBalanceOf } from './queries/getBalanceOf';
export * from './queries/getBalanceOf';
export { default as useGetBalanceOf } from './queries/getBalanceOf/useGetBalanceOf';

export { default as getTokenBalances } from './queries/getTokenBalances';
export * from './queries/getTokenBalances';
export { default as useGetTokenBalances } from './queries/getTokenBalances/useGetTokenBalances';

export { default as getTokenUsdPrice } from './queries/getTokenUsdPrice';
export * from './queries/getTokenUsdPrice';
export { default as useGetTokenUsdPrice } from './queries/getTokenUsdPrice/useGetTokenUsdPrice';

export { default as getVrtConversionEndTime } from './queries/getVrtConversionEndTime';
export * from './queries/getVrtConversionEndTime';
export { default as useGetVrtConversionEndTime } from './queries/getVrtConversionEndTime/useGetVrtConversionEndTime';

export { default as getVrtConversionRatio } from './queries/getVrtConversionRatio';
export * from './queries/getVrtConversionRatio';
export { default as useGetVrtConversionRatio } from './queries/getVrtConversionRatio/useGetVrtConversionRatio';

export { default as getVenusVaiVaultDailyRate } from './queries/getVenusVaiVaultDailyRate';
export * from './queries/getVenusVaiVaultDailyRate';
export { default as useGetVenusVaiVaultDailyRate } from './queries/getVenusVaiVaultDailyRate/useGetVenusVaiVaultDailyRate';

export { default as getXvsWithdrawableAmount } from './queries/getXvsWithdrawableAmount';
export * from './queries/getXvsWithdrawableAmount';

export { default as useGetXvsWithdrawableAmount } from './queries/getXvsWithdrawableAmount/useGetXvsWithdrawableAmount';

export { default as useGetAsset } from './queries/useGetAsset';

export { default as useGetLegacyPoolTotalXvsDistributed } from './queries/useGetLegacyPoolTotalXvsDistributed';

export * from './queries/getLegacyPool';
export { default as useGetLegacyPool } from './queries/getLegacyPool/useGetLegacyPool';

export { default as useGetPools } from './queries/useGetPools';

export { default as useGetPool } from './queries/useGetPool';

export { default as useGetIsolatedPoolsTreasuryTotals } from './queries/useGetIsolatedPoolsTreasuryTotals';

export { default as getIsolatedPools } from './queries/getIsolatedPools';
export * from './queries/getIsolatedPools';
export { default as useGetIsolatedPools } from './queries/getIsolatedPools/useGetIsolatedPools';

export { default as getMarketHistory } from './queries/getMarketHistory';
export * from './queries/getMarketHistory';
export { default as useGetMarketHistory } from './queries/getMarketHistory/useGetMarketHistory';

export { default as getVTokenInterestRateModel } from './queries/getVTokenInterestRateModel';
export * from './queries/getVTokenInterestRateModel';
export { default as useGetVTokenInterestRateModel } from './queries/getVTokenInterestRateModel/useGetVTokenInterestRateModel';

export { default as getVTokenApySimulations } from './queries/getVTokenApySimulations';
export * from './queries/getVTokenApySimulations';
export { default as useGetVTokenApySimulations } from './queries/getVTokenApySimulations/useGetVTokenApySimulations';

export { default as getCurrentVotes } from './queries/getCurrentVotes';
export * from './queries/getCurrentVotes';
export { default as useGetCurrentVotes } from './queries/getCurrentVotes/useGetCurrentVotes';

export { default as getTransactions } from './queries/getTransactions';
export * from './queries/getTransactions';
export { default as useGetTransactions } from './queries/getTransactions/useGetTransactions';

export { default as getXvsVaultPoolCount } from './queries/getXvsVaultPoolCount';
export * from './queries/getXvsVaultPoolCount';
export { default as useGetXvsVaultPoolCount } from './queries/getXvsVaultPoolCount/useGetXvsVaultPoolCount';

export { default as getXvsVaultPoolInfo } from './queries/getXvsVaultPoolInfo';
export * from './queries/getXvsVaultPoolInfo';
export { default as useGetXvsVaultPoolInfo } from './queries/getXvsVaultPoolInfo/useGetXvsVaultPoolInfo';

export { default as getXvsVaultRewardPerBlock } from './queries/getXvsVaultRewardPerBlock';
export * from './queries/getXvsVaultRewardPerBlock';
export { default as useGetXvsVaultRewardPerBlock } from './queries/getXvsVaultRewardPerBlock/useGetXvsVaultRewardPerBlock';

export { default as getXvsVaultTotalAllocationPoints } from './queries/getXvsVaultTotalAllocationPoints';
export * from './queries/getXvsVaultTotalAllocationPoints';
export { default as useGetXvsVaultTotalAllocationPoints } from './queries/getXvsVaultTotalAllocationPoints/useGetXvsVaultTotalAllocationPoints';

export { default as getXvsVaultUserInfo } from './queries/getXvsVaultUserInfo';
export * from './queries/getXvsVaultUserInfo';
export { default as useGetXvsVaultUserInfo } from './queries/getXvsVaultUserInfo/useGetXvsVaultUserInfo';

export { default as getXvsVaultLockedDeposits } from './queries/getXvsVaultLockedDeposits';
export * from './queries/getXvsVaultLockedDeposits';
export { default as useGetXvsVaultLockedDeposits } from './queries/getXvsVaultLockedDeposits/useGetXvsVaultLockedDeposits';

export { default as useGetVaults } from './queries/useGetVaults';

export { default as getProposals } from './queries/getProposals';
export * from './queries/getProposals';
export { default as useGetProposals } from './queries/getProposals/useGetProposals';

export { default as getProposal } from './queries/getProposals/getProposal';
export * from './queries/getProposals/getProposal';
export { default as useGetProposal } from './queries/getProposals/useGetProposal';

export { default as getVoteReceipt } from './queries/getVoteReceipt';
export * from './queries/getVoteReceipt';
export { default as useGetVoteReceipt } from './queries/getVoteReceipt/useGetVoteReceipt';

export { default as getVoters } from './queries/getVoters';
export * from './queries/getVoters';
export { default as useGetVoters } from './queries/getVoters/useGetVoters';

export { default as getVoterDetails } from './queries/getVoterDetails';
export * from './queries/getVoterDetails';
export { default as useGetVoterDetails } from './queries/getVoterDetails/useGetVoterDetails';

export { default as getVoteSummary } from './queries/getVoteSummary';
export * from './queries/getVoteSummary';
export { default as useGetVoteSummary } from './queries/getVoteSummary/useGetVoteSummary';

export { default as getVoterHistory } from './queries/getVoterHistory';
export * from './queries/getVoterHistory';
export { default as useGetVoterHistory } from './queries/getVoterHistory/useGetVoterHistory';

export { default as getVaiVaultUserInfo } from './queries/getVaiVaultUserInfo';
export * from './queries/getVaiVaultUserInfo';
export { default as useGetVaiVaultUserInfo } from './queries/getVaiVaultUserInfo/useGetVaiVaultUserInfo';

export { default as useGetVestingVaults } from './queries/useGetVaults/useGetVestingVaults';

export { default as getVoteDelegateAddress } from './queries/getVoteDelegateAddress';
export * from './queries/getVoteDelegateAddress';
export { default as useGetVoteDelegateAddress } from './queries/getVoteDelegateAddress/useGetVoteDelegateAddress';

export { default as getVoterAccounts } from './queries/getVoterAccounts';
export * from './queries/getVoterAccounts';
export { default as useGetVoterAccounts } from './queries/getVoterAccounts/useGetVoterAccounts';

export { default as getProposalThreshold } from './queries/getProposalThreshold';
export * from './queries/getProposalThreshold';
export { default as useGetProposalThreshold } from './queries/getProposalThreshold/useGetProposalThreshold';

export { default as getProposalState } from './queries/getProposalState';
export * from './queries/getProposalState';
export { default as useGetProposalState } from './queries/getProposalState/useGetProposalState';

export { default as getLatestProposalIdByProposer } from './queries/getLatestProposalIdByProposer';
export * from './queries/getLatestProposalIdByProposer';
export { default as useGetLatestProposalIdByProposer } from './queries/getLatestProposalIdByProposer/useGetLatestProposalIdByProposer';

export { default as getMintableVai } from './queries/getMintableVai';
export * from './queries/getMintableVai';
export { default as useGetMintableVai } from './queries/getMintableVai/useGetMintableVai';

export { default as getBlockNumber } from './queries/getBlockNumber';
export * from './queries/getBlockNumber';
export { default as useGetBlockNumber } from './queries/getBlockNumber/useGetBlockNumber';

export { default as getProposalEta } from './queries/getProposalEta';
export * from './queries/getProposalEta';
export { default as useGetProposalEta } from './queries/getProposalEta/useGetProposalEta';

export { default as getPancakeSwapPairs } from './queries/getPancakeSwapPairs';
export * from './queries/getPancakeSwapPairs';
export { default as useGetPancakeSwapPairs } from './queries/getPancakeSwapPairs/useGetPancakeSwapPairs';

export { default as getVaiRepayApr } from './queries/getVaiRepayApr';
export * from './queries/getVaiRepayApr';
export { default as useGetVaiRepayApr } from './queries/getVaiRepayApr/useGetVaiRepayApr';

export { default as getXvsVaultPendingWithdrawalsFromBeforeUpgrade } from './queries/getXvsVaultPendingWithdrawalsFromBeforeUpgrade';
export * from './queries/getXvsVaultPendingWithdrawalsFromBeforeUpgrade';
export { default as useGetXvsVaultPendingWithdrawalsFromBeforeUpgrade } from './queries/getXvsVaultPendingWithdrawalsFromBeforeUpgrade/useGetXvsVaultPendingWithdrawalsFromBeforeUpgrade';

export { default as getVTokens } from './queries/getVTokens';
export * from './queries/getVTokens';
export { default as useGetVTokens } from './queries/getVTokens/useGetVTokens';

export { default as getPrimeToken } from './queries/getPrimeToken';
export * from './queries/getPrimeToken';
export { default as useGetPrimeToken } from './queries/getPrimeToken/useGetPrimeToken';

export { default as getHypotheticalPrimeApys } from './queries/getHypotheticalPrimeApys';
export * from './queries/getHypotheticalPrimeApys';
export { default as useGetHypotheticalPrimeApys } from './queries/getHypotheticalPrimeApys/useGetHypotheticalPrimeApys';

export { default as getPrimeStatus } from './queries/getPrimeStatus';
export * from './queries/getPrimeStatus';
export { default as useGetPrimeStatus } from './queries/getPrimeStatus/useGetPrimeStatus';

export * from './queries/getLatestAppVersion';
export { default as useGetLatestAppVersion } from './queries/getLatestAppVersion/useGetLatestAppVersion';

export { default as useGetPrimeEstimation } from './queries/useGetPrimeEstimation';

export { default as getPrimeDistributionForMarket } from './queries/getPrimeDistributionForMarket';
export * from './queries/getPrimeDistributionForMarket';
export { default as useGetPrimeDistributionForMarket } from './queries/getPrimeDistributionForMarket/useGetPrimeDistributionForMarket';

export * from './queries/getVaiVaultPaused';
export { default as useGetVaiVaultPaused } from './queries/getVaiVaultPaused/useGetVaiVaultPaused';

export * from './queries/getXvsVaultPaused';
export { default as useGetXvsVaultPaused } from './queries/getXvsVaultPaused/useGetXvsVaultPaused';

export { default as getXvsBridgeFeeEstimation } from './queries/getXvsBridgeFeeEstimation';
export * from './queries/getXvsBridgeFeeEstimation';
export { default as useGetXvsBridgeFeeEstimation } from './queries/getXvsBridgeFeeEstimation/useGetXvsBridgeFeeEstimation';

export { default as getXvsBridgeStatus } from './queries/getXvsBridgeStatus';
export * from './queries/getXvsBridgeStatus';
export { default as useGetXvsBridgeStatus } from './queries/getXvsBridgeStatus/useGetXvsBridgeStatus';

export { default as getXvsBridgeMintStatus } from './queries/getXvsBridgeMintStatus';
export * from './queries/getXvsBridgeMintStatus';
export { default as useGetXvsBridgeMintStatus } from './queries/getXvsBridgeMintStatus/useGetXvsBridgeMintStatus';

export { default as getPoolDelegateApprovalStatus } from './queries/getPoolDelegateApprovalStatus';
export * from './queries/getPoolDelegateApprovalStatus';
export { default as useGetPoolDelegateApprovalStatus } from './queries/getPoolDelegateApprovalStatus/useGetPoolDelegateApprovalStatus';
