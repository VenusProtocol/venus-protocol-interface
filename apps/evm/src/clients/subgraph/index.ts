export * from './queries/isolatedPools/getIsolatedPoolParticipantsCount';
export * from './queries/governance/getProposals';
