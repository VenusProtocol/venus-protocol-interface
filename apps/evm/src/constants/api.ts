import type { Network } from 'types';

export const apiUrls: Record<Network, string> = {
  testnet: 'https://testnetapi.venus.io',
  mainnet: 'https://api.venus.io',
  'mainnet-preview': 'https://api-preview.venus.io',
};
