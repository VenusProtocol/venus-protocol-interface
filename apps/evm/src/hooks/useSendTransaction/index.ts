import { ContractReceipt, ContractTransaction } from 'ethers';
import { MutationObserverOptions, useMutation } from 'react-query';

import FunctionKey from 'constants/functionKey';
import { TransactionType } from 'types';

import { CONFIRMATIONS, useTrackTransaction } from './useTrackTransaction';

export interface UseSendTransactionOptions<TMutateInput extends Record<string, unknown> | void>
  extends MutationObserverOptions<unknown, Error, TMutateInput> {
  waitForConfirmation?: boolean;
}

export interface UseSendTransactionInput<TMutateInput extends Record<string, unknown> | void> {
  fn: (input: TMutateInput) => Promise<ContractTransaction>;
  fnKey: FunctionKey | [FunctionKey, ...unknown[]];
  transactionType?: TransactionType;
  onConfirmed?: (input: {
    transaction: ContractTransaction;
    transactionReceipt: ContractReceipt;
    input: TMutateInput;
  }) => Promise<unknown> | unknown;
  onReverted?: (input: {
    transaction: ContractTransaction;
    input: TMutateInput;
  }) => Promise<unknown> | unknown;
  options?: UseSendTransactionOptions<TMutateInput>;
}

export const useSendTransaction = <TMutateInput extends Record<string, unknown> | void>({
  fn,
  fnKey,
  transactionType,
  onConfirmed,
  onReverted,
  options,
}: UseSendTransactionInput<TMutateInput>) => {
  const trackTransaction = useTrackTransaction({ transactionType });

  return useMutation(
    fnKey,
    async input => {
      // Send transaction
      const transaction = await fn(input);

      // Track transaction's progress in the background
      trackTransaction({
        transaction,
        onConfirmed: onConfirmedInput => onConfirmed?.({ ...onConfirmedInput, input }),
        onReverted: onRevertedInput => onReverted?.({ ...onRevertedInput, input }),
      });

      if (options?.waitForConfirmation) {
        // Only return when transaction has been confirmed
        await transaction.wait(CONFIRMATIONS);
      }
    },
    options,
  );
};
