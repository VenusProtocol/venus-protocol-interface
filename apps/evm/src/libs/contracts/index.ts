export * from './generated/addresses';
export * from './generated/abis';

export * from './utilities/getContractAddress';
