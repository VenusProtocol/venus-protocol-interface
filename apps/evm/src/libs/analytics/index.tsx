import { PostHogProvider, usePostHog } from 'posthog-js/react';
import { useEffect } from 'react';

import config from 'config';
import { version as APP_VERSION } from 'constants/version';
import { useAccountAddress, useChainId } from 'libs/wallet';

const UserIdentifier: React.FC = () => {
  const { accountAddress } = useAccountAddress();
  const { chainId } = useChainId();
  const posthog = usePostHog();

  // Identify user by their account address along with the network chain they are on
  useEffect(() => {
    if (accountAddress) {
      posthog?.identify(accountAddress, { chainId });
    } else {
      posthog?.reset();
    }
  }, [accountAddress, chainId, posthog]);

  return null;
};

export interface AnalyticProviderProps {
  children?: React.ReactNode;
}

export const AnalyticProvider: React.FC<AnalyticProviderProps> = ({ children }) => (
  <PostHogProvider
    apiKey={config.posthog.apiKey}
    options={{
      api_host: config.posthog.hostUrl,
      persistence: 'memory',
      name: APP_VERSION,
    }}
  >
    <UserIdentifier />

    {children}
  </PostHogProvider>
);

export * from './useAnalytics';
export { default as useAnalytics } from './useAnalytics';
