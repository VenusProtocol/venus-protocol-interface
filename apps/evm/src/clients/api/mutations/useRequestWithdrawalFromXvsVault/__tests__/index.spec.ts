import fakeAccountAddress, {
  altAddress as xvsVaultContractAddress,
} from '__mocks__/models/address';
import { xvs } from '__mocks__/models/tokens';
import { queryClient } from 'clients/api';
import { useSendTransaction } from 'hooks/useSendTransaction';
import { useAnalytics } from 'libs/analytics';
import { useGetXvsVaultContractAddress } from 'libs/contracts';
import { useGetToken } from 'libs/tokens';
import { renderHook } from 'testUtils/render';
import type { Address } from 'viem';
import type { Mock } from 'vitest';
import { useRequestWithdrawalFromXvsVault } from '..';

vi.mock('libs/analytics');
vi.mock('libs/contracts');

const fakeInput = {
  rewardTokenAddress: '0x8301F2213c0eeD49a7E28Ae4c3e91722919B8B47' as Address,
  poolIndex: 4n,
  amountMantissa: 1000000000000n,
};

const fakeOptions = {
  gasless: true,
  waitForConfirmation: true,
};

describe('useRequestWithdrawalFromXvsVault', () => {
  beforeEach(() => {
    (useGetXvsVaultContractAddress as Mock).mockImplementation(() => xvsVaultContractAddress);
    (useGetToken as Mock).mockImplementation(() => xvs);
  });

  it('calls useSendTransaction with the correct parameters', async () => {
    const mockCaptureAnalyticEvent = vi.fn();
    (useAnalytics as Mock).mockImplementation(() => ({
      captureAnalyticEvent: mockCaptureAnalyticEvent,
    }));

    renderHook(() => useRequestWithdrawalFromXvsVault(fakeOptions), {
      accountAddress: fakeAccountAddress,
    });

    expect(useSendTransaction).toHaveBeenCalledWith({
      fn: expect.any(Function),
      onConfirmed: expect.any(Function),
      options: fakeOptions,
    });

    const { fn, onConfirmed } = (useSendTransaction as Mock).mock.calls[0][0];

    expect(await fn(fakeInput)).toMatchInlineSnapshot(`
      {
        "abi": [],
        "address": "0xa258a693A403b7e98fd05EE9e1558C760308cFC7",
        "args": [
          "0x8301F2213c0eeD49a7E28Ae4c3e91722919B8B47",
          4n,
          1000000000000n,
        ],
        "functionName": "requestWithdrawal",
      }
    `);

    onConfirmed({ input: fakeInput });

    expect(mockCaptureAnalyticEvent).toHaveBeenCalledTimes(1);
    expect(mockCaptureAnalyticEvent.mock.calls[0]).toMatchInlineSnapshot(`
      [
        "Token withdrawal requested from XVS vault",
        {
          "poolIndex": 4,
          "rewardTokenSymbol": "XVS",
          "tokenAmountTokens": 0.000001,
        },
      ]
    `);

    expect(queryClient.invalidateQueries).toHaveBeenCalledTimes(2);
    expect((queryClient.invalidateQueries as Mock).mock.calls).toMatchSnapshot();
  });

  it('throws error when XVS Vault contract address is not found', async () => {
    (useGetXvsVaultContractAddress as Mock).mockImplementation(() => undefined);

    renderHook(() => useRequestWithdrawalFromXvsVault(fakeOptions));

    const { fn } = (useSendTransaction as Mock).mock.calls[0][0];

    await expect(async () => fn(fakeInput)).rejects.toThrow('somethingWentWrong');
  });
});
