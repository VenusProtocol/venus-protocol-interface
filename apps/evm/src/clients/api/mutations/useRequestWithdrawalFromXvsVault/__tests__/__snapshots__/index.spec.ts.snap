// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useRequestWithdrawalFromXvsVault > calls useSendTransaction with the correct parameters 3`] = `
[
  [
    {
      "queryKey": [
        "GET_XVS_VAULT_USER_INFO",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "poolIndex": 4,
          "rewardTokenAddress": "0xB9e0E753630434d7863528cc73CB7AC638a7c8ff",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_XVS_VAULT_WITHDRAWAL_REQUESTS",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "poolIndex": 4,
          "rewardTokenAddress": "0xB9e0E753630434d7863528cc73CB7AC638a7c8ff",
        },
      ],
    },
  ],
]
`;
